\documentclass{article}
\usepackage[sc]{mathpazo}
\usepackage[T1]{fontenc}
\usepackage{geometry}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{float}
\usepackage{caption}
\usepackage{booktabs}
\geometry{verbose,tmargin=2.5cm,bmargin=2.5cm,lmargin=2.5cm,rmargin=2.5cm}
\setcounter{secnumdepth}{2}
\setcounter{tocdepth}{2}
\usepackage{url}
\usepackage[unicode=true,pdfusetitle,
 bookmarks=true,bookmarksnumbered=true,bookmarksopen=true,bookmarksopenlevel=2,
 breaklinks=false,pdfborder={0 0 1},backref=false,colorlinks=false]
 {hyperref}
\hypersetup{
 pdfstartview={XYZ null null 1}}
\usepackage{breakurl}
\usepackage{array}
\newcommand\Mark[1]{\textsuperscript#1}
\renewcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textbf{#1}}}%

\begin{document}

<<setup, include=FALSE, cache=FALSE>>=
# set global chunk options
# for figures
opts_chunk$set(fig.path='figs/', fig.align='center', fig.show='hold',
               dev='CairoPDF', out.width='.4\\linewidth')
# replacing "=" into "->" to make it R thing
options(replace.assign=TRUE,width=90)
# caching chunks
opts_chunk$set(cache.extra = R.version,cache.path='cache/')
opts_chunk$set(cache.extra = rand_seed)
@

\begingroup
\centering
{\LARGE Predicting the Chromatin Conformation From ChIPseq Dataset: part 2 (Motif Discovery Analysis) \\[1.5em]
\large Ricky Lim\Mark{1}, Samuel Collombet\Mark{2}, Agus Salim\Mark{3}, Touati Benoukraf\Mark{1}}\\[1em]
\begin{tabular}{*{3}{>{\centering}p{.25\textwidth}}}
    \Mark{1}Cancer Science Institute of Singapore, National University of Singapore&\Mark{2}Institut de Biologie de l'Ecole Normale Superieur de Paris& \Mark{3}Department of Mathematics and Statistics& \tabularnewline
%National University of Singapore& La Trobe University& National University of Singapore\tabularnewline
%\url{csilr@nus.edu.sg} & \url{samuel.collombet@ens.fr} & \url{A.Salim@latrobe.edu.au} & \url{csitb@nus.edu.sg}
\url{benoukraf@nus.edu.sg}
\end{tabular}\par
\endgroup

\begin{verbatim}
  Filename: motifAnalysis.Rnw 
  Working directory: \Sexpr{getwd()} 
\end{verbatim}

<<Rcodes>>=
source('/home/ricky/Rlim/ChromatinConformation/MotifCalls/MotifCalls.R')
work_dir= '/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/'
@

\section{Introduction}
Goal: to infer the motif bindings of transcription factors and co-factors from ChIPseq experiment.

\section{Local Clustering: Direct and Indirect Bindings}
Here, we applied local clustering following the component calls by negative binomial mixture models. 
The distance of 3Kb, 8Kb, 20Kb, and 50Kb was used for this clustering.\\

With local clustering, we found that most of the clusters are single peaks. 
Increasing the clustering distance from 3Kb to 50Kb gives rise to the decreasing coverage percentage of clusters with single peaks.
Although the majority of the clusters remains single peaks (figure ~\ref{fig:numPeaks}).

As we are interested in the chromatin conformation consisting of direct transcription factors and cofactors, 
we excluded the clusters with only single peaks.

<<DataPrep, echo=FALSE, cache=TRUE>>=

Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist3kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance=3000, filterOutSingle=FALSE, 
    output_f = paste0(work_dir,
    'Output/Koeffler_BM_CebpE_NBM_BiclusterAssignment_compSorted4_dist3kb'))

Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist5kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance=5000,filterOutSingle=FALSE,
    paste0(work_dir,
    'Output/Koeffler_BM_CebpE_NBM_BiclusterAssignment_compSorted4_dist5kb'))

Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist8kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance=8000,filterOutSingle=FALSE,
    paste0(work_dir,
    'Output/Koeffler_BM_CebpE_NBM_BiclusterAssignment_compSorted4_dist8kb'))

Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist20kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance=20000,filterOutSingle=FALSE,
    paste0(work_dir,
    'Output/Koeffler_BM_CebpE_NBM_BiclusterAssignment_compSorted4_dist20kb'))

Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist50kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance=50000,filterOutSingle=FALSE,
    paste0(work_dir,
    'Output/Koeffler_BM_CebpE_NBM_BiclusterAssignment_compSorted4_dist50kb'))

# filter out the clusters with single peaks only
Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist3kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance = 3000, filterOutSingle = TRUE, 
    paste0(work_dir,
    'Output/',
    'Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist3kb'))
Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist5kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance = 5000, filterOutSingle = TRUE, 
    paste0(work_dir,
    'Output/',
    'Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist5kb'))

Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist8kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance = 8000, filterOutSingle = TRUE, 
    paste0(work_dir,
    'Output/',
    'Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist8kb'))

Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist20kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance = 20000, filterOutSingle = TRUE, 
    paste0(work_dir,
    'Output/',
    'Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist20kb'))

Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist50kb <-
 assignBiclustering(paste0(work_dir,
    'Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed'),
    distance = 50000, filterOutSingle = TRUE, 
    paste0(work_dir,
    'Output/',
    'Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb'))
@

<<DataExplorePlot, echo=FALSE, results='hide', cache=TRUE>>=

plotCountCluster(Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist3kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist3kb', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist3kb_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist5kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist5kb', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist5kb_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist8kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist8kb', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist8kb_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist20kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist20kb', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist20kb_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist50kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist50kb', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist50kb_countCluster.pdf'))

# Singlepeaks filteredout
plotCountCluster(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist3kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist3kb: SinglePeaks Filteredout', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist3kb_SinglePeakFilteredOut_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist5kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist50kb: SinglePeaks Filteredout', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist5kb_SinglePeakFilteredOut_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist8kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist50kb: SinglePeaks Filteredout', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist8kb_SinglePeakFilteredOut_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist20kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist50kb: SinglePeaks Filteredout', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist20kb_SinglePeakFilteredOut_countCluster.pdf'))
plotCountCluster(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist50kb$result$local, 
        title_p = 'Koeffler_BM_CebpE_NBM_comp4_dist50kb: SinglePeaks Filteredout', 
        output_f = paste0(work_dir, 
        'Output/Koeffler_BM_CebpE_NBM_comp4_dist50kb_SinglePeakFilteredOut_countCluster.pdf'))

@
<<DataExploreTable>>=

## Table
Koeffler_BM_CebpE_Bicluster_NBM_table <- matrix(nrow=4, ncol=5) 
rownames(Koeffler_BM_CebpE_Bicluster_NBM_table) <- c('Direct', 'Indirect', 
                                                     'Noise', 'Total')
colnames(Koeffler_BM_CebpE_Bicluster_NBM_table) <- c('Dist3kb', 'Dist5kb', 'Dist8kb', 
                                                 'Dist20kb', 'Dist50kb')
Koeffler_BM_CebpE_Bicluster_NBM_table[, 1] <- 
    Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist3kb$numbindings 
Koeffler_BM_CebpE_Bicluster_NBM_table[, 2] <- 
    Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist5kb$numbindings 
Koeffler_BM_CebpE_Bicluster_NBM_table[, 3] <- 
    Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist8kb$numbindings 
Koeffler_BM_CebpE_Bicluster_NBM_table[, 4] <- 
    Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist20kb$numbindings 
Koeffler_BM_CebpE_Bicluster_NBM_table[, 5] <- 
    Koeffler_BM_CebpE_Bicluster_NBM_comp4_dist50kb$numbindings 

## Table: single peaks filtered out
Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table <- matrix(nrow=4, ncol=5) 
rownames(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table) <- 
    c('Direct', 'Indirect', 'Noise', 'Total')
colnames(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table) <- 
    c('Dist3kb', 'Dist5kb', 'Dist8kb', 'Dist20kb', 'Dist50kb')

Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table[, 1] <- 
    Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist3kb$numbindings 
Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table[, 2] <- 
    Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist5kb$numbindings 
Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table[, 3] <- 
    Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist8kb$numbindings 
Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table[, 4] <- 
    Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist20kb$numbindings 
Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table[, 5] <- 
    Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_comp4_dist50kb$numbindings 

@



\clearpage
\subsection{The Number of Peaks in Clusters}

\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 3 in, height = 3 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist3kb_countCluster.pdf}}
    \subfloat{\includegraphics[width = 3 in, height = 3 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist8kb_countCluster.pdf}}
    \\
    \subfloat{\includegraphics[width = 3 in, height = 3 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist20kb_countCluster.pdf}}
    \subfloat{\includegraphics[width = 3 in, height = 3 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist50kb_countCluster.pdf}}
    \caption{The Number of Peaks (percentage) in Clusters}
    \label{fig:numPeaks}
\end{figure}
\clearpage

\subsection{The Number of Peaks in Cluster: Singlepeaks Filteredout}

\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 2 in, height = 2 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist3kb_SinglePeakFilteredOut_countCluster.pdf}}
    \subfloat{\includegraphics[width = 2 in, height = 2 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist5kb_SinglePeakFilteredOut_countCluster.pdf}}
    \subfloat{\includegraphics[width = 2 in, height = 2 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist8kb_SinglePeakFilteredOut_countCluster.pdf}}
    \\
    \subfloat{\includegraphics[width = 2 in, height = 2 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist20kb_SinglePeakFilteredOut_countCluster.pdf}}
    \subfloat{\includegraphics[width = 2 in, height = 2 in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Koeffler_BM_CebpE_NBM_comp4_dist50kb_SinglePeakFilteredOut_countCluster.pdf}}
    \caption{The Number of Peaks (percentage) in Clusters without Single Peaks}
    \label{fig:numPeaksNoSingleCell}
\end{figure}
\clearpage

<<biclusteringNBMTable, echo=FALSE, results='asis'>>=

print(xtable(Koeffler_BM_CebpE_Bicluster_NBM_table, 
             caption='Clustering Peaks in Different Cluster Distance'))
print(xtable(Koeffler_BM_CebpE_BiclusterSinglePeakFilteredOut_NBM_table, 
             caption='Clustering Peaks in Different Cluster Distance: Single Peaks Filteredout'))
@
\clearpage

\section{Database-based Motif Discovery: Centdist}



<<MotifDataPrep, cache=TRUE>>=
# awk -F'\t' '{print $1"\t"$2-500"\t"$3+500"\t"$5}' 
# Input/Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed 
# > Output/Koeffler_BM_CebpE_Peaks_1kb.bed
@

We run CentDist to search for the occurrances of database motifs (JASPAR and TRANSFAC) \href{biogpu.ddns.comp.nus.edu.sg}{CentDist} on the ChIPseq Peaks indirect and direct clusters.
In all cluster distance, however the CEBP motifs were found in both direct and indirect clusters. 

\begin{verbatim}
/home/ricky/Rlim/ChromatinConformation/MotifCalls/P53/Output/CentDist

# change the '_' and '$' into '-' as these characters were already assigned in LaTeX
sed 's/\_\|\$/-/g' result.txt > tmp
mv tmp result.txt
\end{verbatim}


<<CentDistP53>>=
work_dir_centdist = '/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/'
Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_direct <- 
    read.delim(paste0(work_dir,'Output/CentDist/',
    'Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_direct/',
    'result.txt'))
Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_indirect <- 
    read.delim(paste0(work_dir,'Output/CentDist/',
    'Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_indirect/',
    'result.txt'))
@
\clearpage





\section{Centdist Motif Results for Direct and Indirect Clusters within 50kb: CebpE}
\begin{table}[h!]
  \centering
    \multicolumn{2}{c}{Direct}
    \multicolumn{2}{c}{Indirect} \\
    \cline{1-2}
    \cline{2-4}
  \begin{tabular}{ | m{5cm} | c | m{5cm} | c | }
    \hline
    Info & Dist & Info & Dist\\ \hline

      %row1
      \begin{itemize}
        \item Name: \Sexpr{Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_direct[11,'Name']}
        \item Binding.Range: \Sexpr{Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_direct[11,'Binding.Range']}
        \item P.value: \Sexpr{Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_direct[11,'P.value']}
        \item Rank: 11
      \end{itemize}
      &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/CentDist/Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_direct/V$CEBP_01.png}
    \end{minipage}
    &
      \begin{itemize}
        \item Name: \Sexpr{Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_indirect[32,'Name']}
        \item Binding.Range: \Sexpr{Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_indirect[32,'Binding.Range']}
        \item P.value: \Sexpr{Koeffler_BM_ChIPseq_CebpE_Bicluster_compSorted_dist50kb_indirect[32,'P.value']}
        \item Rank: 32
      \end{itemize}
    &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/CentDist/Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_indirect/V$CEBP_Q2.png}
    \end{minipage}
    \\ \hline

  \end{tabular}
  \caption{CebpE ChIPseq in BM: CentDist Motif Discovery}\label{CebpECentdist}
\end{table}

\clearpage

\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 7in, height = 3in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/CentDist/Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_direct/top5_Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_direct.png}}
    \\
    \subfloat{\includegraphics[width = 7in, height = 3in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/CentDist/Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_indirect/top5_Koeffler_BM_CebpE_NBM_BiclusterAssignment_SinglePeakFilteredOut_compSorted4_dist50kb_indirect.png}}
    \caption{Top 5 Motifs Discovered from Direct and Indirect Clusters.}
    \label{fig:chipseqScoreHomoHeteroCebpMotif}
\end{figure}



\section{De Novo Motif Discovery}

\subsection{PeakMotif: RSAT}
\begin{verbatim}
Command: nice -n 19 /home/rsat/rsat/perl-scripts/peak-motifs  
-v 1 -title 'Koeffler_BM_CebpE_Peak-Motifs' 
-i /home/rsat/rsat/public_html/tmp/wwwrun/2014/11/03/
peak-motifs.2014-11-03.083858_2014-11-03.083858_XPVNnt/peak-motifspeak_seq   
-max_seq_len 1000 -markov auto -disco oligos,dyads,positions,local_words 
-nmotifs 10  -minol 6 -maxol 8  -no_merge_lengths -2str  
-origin center  -motif_db jaspar_core_vertebrates tf 
/home/rsat/rsat/public_html/data/motif_databases/JASPAR/jaspar_core_vertebrates_2013-11.tf 
-scan_markov 1 -task purge,seqlen,composition,disco,collect_motifs,
motifs_vs_motifs,timelog,archive,synthesis,small_summary,motifs_vs_db,scan 
-prefix peak-motifs -noov -img_format png  
-outdir /home/rsat/rsat/public_html/tmp/
wwwrun/2014/11/03/peak-motifs.2014-11-03.083858_2014-11-03.083858_XPVNnt
\end{verbatim}

<<engine='bash'>>=
# pwd: /home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/
#      Rsat/peak-motifs.2014-11-03.083858_2014-11-03.083858_XPVNnt/results/sites

# skip the lines starting with '#' and ';'
# awk 'NF && $1!~/^#|^;/' peak-motifs_all_motifs_seqcoord.tab > peak_motifs_sites.tab

# for the jaspar db
# pwd: /home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/Rsat/
#      peak-motifs.2014-11-03.083858_2014-11-03.083858_XPVNnt/results/discovered_vs_db

# skip the lines starting with '#' and ';'
# awk 'NF && $1!~/^#|^;/' peak-motifs_motifs_vs_db_jaspar_core_vertebrates.tab > 
#   peak-motifs_db_jaspar.tab

@
<<DataPreProcessingMotif, cache=TRUE>>=
#motif_dir = '/home/ricky/Rlim/ChromatinConformation/MotifCalls/'

Koeffler_BM_CebpE_Peaks_MotifSites <- read.table(paste0(work_dir, 
    'Output/Rsat/peak-motifs.2014-11-03.083858_2014-11-03.083858_XPVNnt/results/',
    'sites/peak_motifs_sites.tab'))

# remove the duplicated sites
Koeffler_BM_CebpE_Peaks_MotifSites_Unique <- Koeffler_BM_CebpE_Peaks_MotifSites[
            !duplicated(Koeffler_BM_CebpE_Peaks_MotifSites$V1),] 
Koeffler_BM_CebpE_Peaks_MotifSites_Unique <- 
            Koeffler_BM_CebpE_Peaks_MotifSites_Unique[, c('V1', 'V3')] 

Koeffler_BM_CebpE_Peaks_MotifSites_Unique <- 
            addChrCoordinates(Koeffler_BM_CebpE_Peaks_MotifSites_Unique)

head(Koeffler_BM_CebpE_Peaks_MotifSites_Unique)

# get the peak score
Koeffler_BM_CebpE_Peaks_1kb  <- read.table(paste0(work_dir, 
                                             'Input/Koeffler_BM_CebpE_Peaks_1kb.bed'))
colnames(Koeffler_BM_CebpE_Peaks_1kb) <- c('chr', 'start', 'end', 'score')
Koeffler_BM_CebpE_Motifs<- 
    merge(Koeffler_BM_CebpE_Peaks_MotifSites_Unique, Koeffler_BM_CebpE_Peaks_1kb, 
          by=c('chr', 'end'))
head(Koeffler_BM_CebpE_Motifs)
Koeffler_BM_CebpE_Motifs <- Koeffler_BM_CebpE_Motifs[, 
          c('chr', 'start', 'end', 'score', 'V3')]
colnames(Koeffler_BM_CebpE_Motifs) <- 
    c('chr', 'start', 'end', 'score', 'motifId')

# get the motif name from jaspar database
Koeffler_BM_CebpE_Jaspar <- read.table(paste0(work_dir, 
    'Output/Rsat/peak-motifs.2014-11-03.083858_2014-11-03.083858_XPVNnt/results/',
    '/discovered_vs_db/peak-motifs_db_jaspar.tab'))
Koeffler_BM_CebpE_Jaspar <- Koeffler_BM_CebpE_Jaspar[, c('V1', 'V4')]
colnames(Koeffler_BM_CebpE_Jaspar) <- c('motifId', 'motifName')
Koeffler_BM_CebpE_Motifs <- merge(Koeffler_BM_CebpE_Motifs, Koeffler_BM_CebpE_Jaspar, 
                        by=c('motifId','motifId'))
Koeffler_BM_CebpE_Motifs <- Koeffler_BM_CebpE_Motifs[, 
                        c('chr', 'start', 'end', 'score', 'motifId', 'motifName')]

# get only CEBP motif
CEBP_motif <- Koeffler_BM_CebpE_Motifs$motifName %in% c('CEBPA', 'CEBPB', 'Cebpa')
Koeffler_BM_CebpE_CEBP_motif <- Koeffler_BM_CebpE_Motifs[CEBP_motif, 
                                                          c('motifName', 'score')] 
Koeffler_BM_CebpE_CEBP_motif[, 'motifName'] <- rep('CEBP', 
                                                   nrow(Koeffler_BM_CebpE_CEBP_motif))
Koeffler_BM_CebpE_nonCEBP_motif <- Koeffler_BM_CebpE_Motifs[!CEBP_motif, 
                                                          c('motifName', 'score')] 


@

<<plotNonCEBPMotif, cache=TRUE, results='hide', echo=FALSE>>=
# the whole motifs in one plot
p <- ggplot(Koeffler_BM_CebpE_nonCEBP_motif, 
            aes(score, colour=motifName, group=motifName )) +
        geom_density()
ggsave(file='figs/Koeffler_BM_CebpE_Score_nonCEBP.pdf')

# plot motifs in chunks, each for 10 motifs
plotNMotifScore(Koeffler_BM_CebpE_nonCEBP_motif, 
                output_f ='figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk', 
                N=5)

@

<<plotCEBPMotifRsat, cache=TRUE, results='hide', echo=FALSE>>=
p <- ggplot(Koeffler_BM_CebpE_CEBP_motif, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif')
ggsave(file='figs/Koeffler_BM_CebpE_Score_CEBP.pdf')

@

\begin{figure}[hbtp]
    \includegraphics{figs/Koeffler_BM_CebpE_Score_CEBP.pdf}
    \caption{Distribution of ChIPseq Scores over all CEBP Motifs}
    \label{fig:chipseqScoreNonCEBP}
\end{figure}
\clearpage

\clearpage
\begin{figure}[hbtp]
    \includegraphics{figs/Koeffler_BM_CebpE_Score_nonCEBP.pdf}
    \caption{Distribution of ChIPseq Scores over all Non-CEBP Motifs}
    \label{fig:chipseqScoreNonCEBP}
\end{figure}
\clearpage


\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 1.5in, height = 2in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk1.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk2.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk3.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk4.pdf}}
    \\
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk5.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk6.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk7.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk8.pdf}}
    \\
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk9.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk10.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk11.pdf}}
    \subfloat{\includegraphics[width = 1.5in, height = 2.in]{figs/Koeffler_BM_CebpE_Score_nonCEBP_chunk12.pdf}}
    \\
    %\subfloat{\includegraphics[width = 1.5in, height = 1.5in]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/Output/Plots/Koeffler_BM_CebpE_Score_nonCEBP_chunk13.pdf}}
    \caption{Distribution of ChIPseq Scores over all Non-CEBP Motifs in Chunks}
    \label{fig:chipseqScoreNonCEBPChunks}
\end{figure}
\clearpage

\subsection{Meme-Chip}

We run meme-chip with the input of ChIPseq peaks of width 0.5kb.
Below is the command-line that we run.
\begin{verbatim}
    meme-chip -db ~/Rlim/Biotools/motif_databases/JASPAR_CORE_2014_vertebrates.meme 
    -oc MotifCalls/Output/ChipMeme 
    -index-name Koeffler_BM_CebpE_Peaks_ChipMeme1000bp 
    -meme-mod zoops -meme-minw 4 -meme-maxw 10 -meme-nmotifs 10 
    MotifCalls/Input/Koeffler_BM_CebpE_Peaks_0.5kb.fasta
\end{verbatim}

Following the run, we analyzed the distribution of ChIPseq scores for each discovered motif.
\begin{verbatim}
pwd: /home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/ChipMeme

# File of origin where we downloaded the fasta files:
meme_out/meme.html
For CebpE, only two motifs that are discovered.

# For centrimo we downloaded the gtf file for JUN motif:
file:///home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE
Output/ChipMeme/Koeffler_BM_Cebpe_Peaks_ChipMemeHomeHeteroCebpE
file stored: /home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/
Output/ChipMeme/jun_sites.gtf

\end{verbatim}

<<engine='bash'>>=

## for a list of motifs (10 motifs output of meme)
# for i in $(ls motif_*_fasta.txt); do awk '/^>/' $i | sed 's/^>//g' | awk '{print $1}'> 
# ''`basename $i .txt`.site''; done

## add filename without _fasta.site to each line
# for i in $(ls *.site);do awk '{print $1"\t"FILENAME}' $i | sed 's/_fasta.site//g' > 
# "`basename $i .site`.msite"; done

## concatenate all the motif sites
# cat *.msite > allmotif.msite

## get the peak with width of 0.5kb
# awk -F``\t'' '{print $1``\t''$2-250``\t''$3+249``\t''$5}' 
#  Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed>Koeffler_BM_CebpE_Peaks_0.5kb.bed
# awk -F"\t" '{print $1"\t"$2-500"\t"$3+499"\t"$5}' 
#  Koeffler_BM_CebpE_NBM_ModelAssignment_compSorted4.bed>Koeffler_BM_CebpE_Peaks_1kb.bed

@

<<memeDataOutput, cache=TRUE>>=
# note that ChIP meme uses ChIP regions of 0.5kb
peak_0.5kb <- read.table(paste0(work_dir, 
                                'Input/Koeffler_BM_CebpE_Peaks_0.5kb.bed'))
colnames(peak_0.5kb) <- c('chr', 'start', 'end', 'score')


all_motifs <- read.table(paste0(work_dir, 
                             'Output/ChipMeme/MotifMeme/allmotif.msite'))
head(all_motifs)
cebp_motif <- all_motifs[all_motifs$V2 == 'motif_2',]

allmotif_score <- getPeakScore(all_motifs, peak_0.5kb)
cebpmotif_score <- getPeakScore(cebp_motif, peak_0.5kb)
colnames(allmotif_score) <- c('chr', 'start', 'end', 'id', 'score')
colnames(cebpmotif_score) <- c('chr', 'start', 'end', 'id', 'score')

#motif_1_score <- getPeakScore(motif_1, peak_0.5kb)
#motif_2_score <- getPeakScore(motif_2, peak_0.5kb)
head(allmotif_score)
head(cebpmotif_score)

@

<<plotCEBPMotifMeme, cache=TRUE, results='hide', echo=FALSE>>=
p <- ggplot(cebpmotif_score, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif: Meme-Chip')
ggsave(file='figs/Koeffler_BM_CebpE_Score_CEBPmotif.pdf')

p <- ggplot(allmotif_score, aes(score, colour=id, group=id)) +
        geom_density()  
ggsave(file='figs/Koeffler_BM_CebpE_Score_Allmotif.pdf')
@
\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 7in, height = 4in]{figs/meme_motif.png}}
    \\
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/Koeffler_BM_CebpE_Score_CEBPmotif.pdf}}
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/Koeffler_BM_CebpE_Score_Allmotif.pdf}}
    \caption{Distribution of ChIPseq Scores over Motif1: Meme-Chip}
    \label{fig:chipseqScoreMotif1}
\end{figure}
\clearpage

\subsection{JunMotif from CentriMo}
<<junChIPSites, engine = 'bash'>>=
## convert jun sites in gtf to bed
# ./gtfMeme2Bed.py jun_sites.gtf > jun_sites.bed  

## intersect the bed files with the ChIPseq bed
# bedtools intersect 
# -b /home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/
#    ChipMeme/jun_sites.bed 
# -a /home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Input/
#    Koeffler_BM_CebpE_Peaks_0.5kb.bed 
# -u > /home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/
#    ChipMeme/MotifMeme/jun_ChIPSites.bed
@
<<plotJun>>=

jun_chipScore <- read.table(paste0(work_dir, 'Output/ChipMeme/MotifMeme/jun_ChIPSites.bed'))
head(jun_chipScore)
colnames(jun_chipScore) <- c('chr', 'start', 'end', 'score')
p <- ggplot(jun_chipScore, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of JUN Motif: Meme-Chip')
ggsave(file='figs/Koeffler_BM_CebpE_Score_JUN_MemeChip.pdf')
@

\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 8in, height = 3in]{figs/jun_motif.png}}
    \\
    \subfloat{\includegraphics[width = 5in, height = 3in]{figs/Koeffler_BM_CebpE_Score_JUN_MemeChip.pdf}}
    \caption{Distribution of ChIPseq Scores over Motifs: Centrimo-Chip}
    \label{fig:chipseqScoreMotifCentrimo}
\end{figure}
\clearpage

\section{Perfect Homodimer and Perfect Heterodimer}
Currently, Cebp has been identified to exist in two forms, i.e., homodimer and heterodimer.
The homodimer motif of Cebp: ATTG \_\_ CAAT.\\
The heterodimer motif of Cebp: TGA \_\_ CAAT.\\
\\

In our dataset in bone marrow, we found \textbf{\Sexpr{nHomodimer} homodimer sites} and \textbf{\Sexpr{nHeterodimer} heterodimer sites}. The ChIP score distribution for these homo and heterodimer Cebp motif is shown in figure (~\ref{fig:chipseqScoreHomoHeteroCebpMotif}).

\begin{verbatim}

The motif sites of perfect homodimer and heterodimer of Cebp are selected from centrimo output.
file:///home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/
ChipMeme/centrimo_out/centrimo.html

The matching sequences (for perfect homo and heterodimer) are then stored at:
/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/Output/ChipMeme/MotifMeme/HomoHetero

\end{verbatim}

<<engine='bash'>>=
## grep only unique sites for homo and heterodimer motif sites
# grep -v -x -f cebpPerfectHeterodimer.site cebpPerfectHomodimer.site > 
# cebpPerfectHomodimerOnly.site
# grep -v -x -f cebpPerfectHomodimer.site cebpPerfectHeterodimer.site > 
# cebpPerfectHeterodimerOnly.site

## add the filename to the 2nd column
# for i in $(ls *.site);do awk '{print $1``\t''FILENAME}' $i | sed 's/.site//g' > ```basename $i .site`.msite''; done

## combine the homo and heterodimer motif sites
# cat cebpPerfectHeterodimerOnly.msite cebpPerfectHomodimerOnly.msite > 
# cebpPerfectHomoHeterodimerOnly.msite
@

<<homoheterodimerDataPrep, cache=TRUE>>=
homohetero_dir=paste0('/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpE/',
                      'Output/ChipMeme/MotifMeme/HomoHetero/')

homoheterodimer_msite <- read.table(paste0(homohetero_dir, 
                                    'cebpPerfectHomoHeterodimerOnly.msite' ))

homodimer_motif <- homoheterodimer_msite[
                        homoheterodimer_msite$V2 == 'cebpPerfectHomodimerOnly',]
nHomodimer <- nrow(homodimer_motif)
heterodimer_motif <- homoheterodimer_msite[
                        homoheterodimer_msite$V2 == 'cebpPerfectHeterodimerOnly',]
nHeterodimer <- nrow(heterodimer_motif)


homoheterodimer_score <- getPeakScore(homoheterodimer_msite, peak_0.5kb)
homodimer_score <- getPeakScore(homodimer_motif, peak_0.5kb)
heterodimer_score<- getPeakScore(heterodimer_motif, peak_0.5kb)
colnames(homoheterodimer_score) <- c('chr', 'start', 'end', 'id', 'score')
colnames(homodimer_score) <- c('chr', 'start', 'end', 'id', 'score')
colnames(heterodimer_score) <- c('chr', 'start', 'end', 'id', 'score')

@


<<plotHomoHeteroCebp, cache=TRUE, results='hide', echo=FALSE>>=
p <- ggplot(homodimer_score, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of Perfect Homodimer CEBP Motif')
ggsave(file='figs/Koeffler_BM_CebpE_Score_PerfectHomoCebp.pdf')

p <- ggplot(heterodimer_score, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of Perfect Heterodimer CEBP Motif')
ggsave(file='figs/Koeffler_BM_CebpE_Score_PerfectHeteroCebp.pdf')

p <- ggplot(homoheterodimer_score, aes(score, colour=id, group=id)) +
        geom_density()  
ggsave(file='figs/Koeffler_BM_CebpE_Score_HomoHeteroCebp.pdf')
@
\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 5in, height = 3in]{figs/Koeffler_BM_CebpE_Score_HomoHeteroCebp.pdf}}
    \\
    \subfloat{\includegraphics[width = 5in, height = 3in]{figs/Koeffler_BM_CebpE_Score_PerfectHomoCebp.pdf}}
    \\
    \subfloat{\includegraphics[width = 5in, height = 3in]{figs/Koeffler_BM_CebpE_Score_PerfectHeteroCebp.pdf}}
    \caption{Distribution of ChIPseq Scores over Homo and Heterodimer of Cebp Motifs.}
    \label{fig:chipseqScoreHomoHeteroCebpMotif}
\end{figure}
\clearpage


\section{Summary}
Our preliminary results from motif discovery analysis show:
\begin{itemize}
    \item The presence of CEBP motif in direct and indirect clusters (CentDist).
    \item CEBP motif is distributed across the range of ChIPseq scores.
    \item Homodimer CEBP motif (\Sexpr{nHomodimer}) occurs more frequently than its heterodimer form (\Sexpr{nHeterodimer}).
    \item CEBP homo and heterodimer motifs show overlapping distribution of ChIPseq scores.
\end{itemize}

The presence of CEBP motif in all clusters (direct and indirect) and across ChIPseq scores could be due to the heterogeneity of cells in our dataset which originated from bone marrow.

\section{Metainfo}
<<>>=
sessionInfo()
@

<<knitIt, cache=TRUE, results='hide', message=FALSE, warning=FALSE>>=
library(knitr)
purl("motifAnalysis.Rnw" ) # compile to tex
purl("motifAnalysis.Rnw", documentation = 0) # extract R code only
knit2pdf("motifAnalysis.Rnw")
@

\end{document}

