\documentclass{article}
\usepackage[sc]{mathpazo}
\usepackage[T1]{fontenc}
\usepackage{geometry}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{float}
\usepackage{caption}
\usepackage{booktabs}
\geometry{verbose,tmargin=2.5cm,bmargin=2.5cm,lmargin=2.5cm,rmargin=2.5cm}
\setcounter{secnumdepth}{2}
\setcounter{tocdepth}{2}
\usepackage{url}
\usepackage[unicode=true,pdfusetitle,
 bookmarks=true,bookmarksnumbered=true,bookmarksopen=true,bookmarksopenlevel=2,
 breaklinks=false,pdfborder={0 0 1},backref=false,colorlinks=false]
 {hyperref}
\hypersetup{
 pdfstartview={XYZ null null 1}}
\usepackage{breakurl}
\usepackage{array}
\newcommand\Mark[1]{\textsuperscript#1}
\renewcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textbf{#1}}}%

\begin{document}

<<setup, include=FALSE, cache=FALSE>>=
# set global chunk options
# for figures
opts_chunk$set(fig.path='figs/', fig.align='center', fig.show='hold',
               dev='CairoPDF', out.width='.4\\linewidth')
# replacing "=" into "->" to make it R thing
options(replace.assign=TRUE,width=90)
# caching chunks
opts_chunk$set(cache.extra = R.version,cache.path='cache/')
opts_chunk$set(cache.extra = rand_seed)
@

\begingroup
\centering
{\LARGE Predicting the Chromatin Conformation From ChIPseq Dataset: part 2 (Motif Discovery Analysis) \\[1.5em]
\large Ricky Lim\Mark{1}, Samuel Collombet\Mark{2}, Agus Salim\Mark{3}, Touati Benoukraf\Mark{1}}\\[1em]
\begin{tabular}{*{3}{>{\centering}p{.25\textwidth}}}
    \Mark{1}Cancer Science Institute of Singapore, National University of Singapore&\Mark{2}Institut de Biologie de l'Ecole Normale Superieur de Paris& \Mark{3}Department of Mathematics and Statistics& \tabularnewline
\url{benoukraf@nus.edu.sg}
\end{tabular}\par
\endgroup

\begin{verbatim}
  Filename: motifAnalysis.Rnw 
  Working directory: \Sexpr{getwd()} 
\end{verbatim}

<<Rcode>>=
source('/home/ricky/Rlim/ChromatinConformation/MotifCalls/MotifCalls.R')
work_dir = '/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/'
@


\section{Database-based Motif Discovery: Centdist}
We run CentDist to search for the occurrances of database motifs (JASPAR and TRANSFAC) \href{biogpu.ddns.comp.nus.edu.sg}{CentDist} on the ChIPseq Peaks indirect and direct clusters.
In all cluster distance, however the CEBP motifs were found in both direct and indirect clusters (in their top 50).\\

OutputDir for cluster distance of 3kb and 50kb:
\begin{verbatim}
/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/Output/CentDist

# change the '_' and '$' into '-' as these characters were already assigned in LaTeX
sed 's/\_\|\$/-/g' result.txt > tmp
mv tmp result.txt
\end{verbatim}


<<CentDistP53>>=
work_dir_centdist = '/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/'
HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_direct <- 
    read.delim(paste0(work_dir_centdist,'Output/CentDist/',
    'HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_direct/',
    'result.txt'))
HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_indirect <- 
    read.delim(paste0(work_dir_centdist,'Output/CentDist/',
    'HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_indirect/',
    'result.txt'))

#HCT116
HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_direct <- 
    read.delim(paste0(work_dir_centdist,'Output/CentDist/',
    'HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_direct/',
    'result.txt'))

HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_indirect <- 
    read.delim(paste0(work_dir_centdist,'Output/CentDist/',
    'HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_indirect/',
    'result.txt'))

#K562
HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_direct <- 
    read.delim(paste0(work_dir_centdist,'Output/CentDist/',
    'HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_direct/',
    'result.txt'))

HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_indirect <- 
    read.delim(paste0(work_dir_centdist,'Output/CentDist/',
    'HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_indirect/',
    'result.txt'))
@
\clearpage

\section{Centdist Motif Results for Direct and Indirect Clusters within 50kb: CebpB}
\begin{table}[h!]
  \centering
    \multicolumn{2}{c}{Direct}
    \multicolumn{2}{c}{Indirect} \\
    \cline{1-2}
    \cline{2-4}
  \begin{tabular}{ | m{5cm} | c | m{5cm} | c | }
    \hline
    Info & Dist & Info & Dist\\ \hline

      %row1
      \begin{itemize}
        \item Name: \Sexpr{HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_direct[15,'Name']}
        \item Binding.Range: \Sexpr{HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_direct[15,'Binding.Range']}
        \item P.value: \Sexpr{HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_direct[15,'P.value']}
        \item Rank: 15
        \item Cell: GM12878
      \end{itemize}
      &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/Output/CentDist/HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_direct/V$CEBPB_02.png}
    \end{minipage}
    &
      \begin{itemize}
        \item Name: \Sexpr{HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_indirect[16,'Name']}
        \item Binding.Range: \Sexpr{HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_indirect[16,'Binding.Range']}
        \item P.value: \Sexpr{HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_indirect[16,'P.value']}
        \item Rank: 16
        \item Cell: GM12878
      \end{itemize}
      &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/Output/CentDist/HudsonAlpha_GM12878_CebpB_Bicluster_compSorted4_dist50kb_indirect/V$CEBPB_02.png}
    \end{minipage}
    \\ \hline

      %row2
      \begin{itemize}
        \item Name: \Sexpr{HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_direct[4,'Name']}
        \item Binding.Range: \Sexpr{HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_direct[4,'Binding.Range']}
        \item P.value: \Sexpr{HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_direct[4,'P.value']}
        \item Rank: 4
        \item Cell: HCT116
      \end{itemize}
      &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/Output/CentDist/HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_direct/V$CEBPB_02.png}
    \end{minipage}
    &
      \begin{itemize}
        \item Name: \Sexpr{HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_indirect[3,'Name']}
        \item Binding.Range: \Sexpr{HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_indirect[3,'Binding.Range']}
        \item P.value: \Sexpr{HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_indirect[3,'P.value']}
        \item Rank: 3
        \item Cell: HCT116
      \end{itemize}
      &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/Output/CentDist/HudsonAlpha_HCT116_CebpB_Bicluster_compSorted4_dist50kb_indirect/V$CEBPB_02.png}
    \end{minipage}
    \\ \hline

      %row3
      \begin{itemize}
        \item Name: \Sexpr{HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_direct[1,'Name']}
        \item Binding.Range: \Sexpr{HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_direct[1,'Binding.Range']}
        \item P.value: \Sexpr{HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_direct[1,'P.value']}
        \item Rank: 1
        \item Cell: K562
      \end{itemize}
      &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/Output/CentDist/HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_direct/V$CEBPB_02.png}
    \end{minipage}
    &
      \begin{itemize}
        \item Name: \Sexpr{HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_indirect[1,'Name']}
        \item Binding.Range: \Sexpr{HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_indirect[1,'Binding.Range']}
        \item P.value: \Sexpr{HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_indirect[1,'P.value']}
        \item Rank: 1
        \item Cell: K562
      \end{itemize}
      &
    \begin{minipage}{.2\textwidth}
      \includegraphics[width=30mm, height=30mm]{/home/ricky/Rlim/ChromatinConformation/MotifCalls/CebpB/Output/CentDist/HudsonAlpha_K562_CebpB_Bicluster_compSorted4_dist50kb_indirect/V$CEBPB_02.png}
    \end{minipage}
    \\ \hline

  \end{tabular}
  \caption{CebpB ChIPseq in BM: CentDist Motif Discovery}\label{CebpBCentdist}
\end{table}

\clearpage

\section{DeNovo Motif Discovery: ChIP-Meme}
\begin{verbatim}
# run chip-meme
meme-chip -db ~/Rlim/Biotools/motif_databases/JASPAR_CORE_2014_vertebrates.meme 
-oc Output/ChipMeme_HudsonAlpha_GM12878 -index-name HudsonAlpha_GM12878_CebpB 
-meme-mod zoops -meme-minw 4 -meme-maxw 10 -meme-nmotifs 10 
Input/HudsonAlpha_GM12878_CebpB_ModelAssignment_compSorted4_0.5kb.fasta  &
meme-chip -db ~/Rlim/Biotools/motif_databases/JASPAR_CORE_2014_vertebrates.meme 
-oc Output/ChipMeme_HudsonAlpha_HCT116 -index-name HudsonAlpha_HCT116_CebpB 
-meme-mod zoops -meme-minw 4 -meme-maxw 10 -meme-nmotifs 10 Input/HudsonAlpha_HCT116_CebpB_ModelAssignment_compSorted4_0.5kb.fasta  &
meme-chip -db ~/Rlim/Biotools/motif_databases/JASPAR_CORE_2014_vertebrates.meme 
-oc Output/ChipMeme_HudsonAlpha_K562 -index-name HudsonAlpha_K562_CebpB 
-meme-mod zoops -meme-minw 4 -meme-maxw 10 -meme-nmotifs 
10 Input/HudsonAlpha_K562_CebpB_ModelAssignment_compSorted4_0.5kb.fasta  &

# process the chip-meme file from meme_out/meme.html, download the discovered motif (fasta)
# prepare for the allmotif_f
for i in $(ls motif_*_fasta.txt); do awk '/^>/' $i | sed 's/^>//g' | awk '{print $1}' >
```basename $i .txt`.site''; done

for i in $(ls *.site);do awk '{print $1``\t''FILENAME}' $i | sed 's/_fasta.site//g' > 
```basename $i .site`.msite''; done

cat *.msite > allmotif.msite

# in Input/
# prepare the peakScore file
awk -F``\t'' '{print $1``\t''$2-250``\t''$3+250``\t''$5}' 
HudsonAlpha_GM12878_CebpB_ModelAssignment_compSorted4.bed > 
HudsonAlpha_GM12878_CebpB_Peaks_0.5kb.bed

\end{verbatim}

\section{DeNovo Motif Discovery: Meme-Chip}

\subsection{Meme-Chip:GM12878}
<<MemeGM12878>>=
meme_chipScore_GM12878 <- getCebpAllMotif(peakScore_f=paste0(work_dir,
                                          'Input/HudsonAlpha_GM12878_CebpB_Peaks_0.5kb.bed'), 
                                          allmotif_f=paste0(work_dir,
                                          'Output/ChipMeme_HudsonAlpha_GM12878/Motif/allmotif.msite'),
                                          cebp_motif= 'motif_3')
head(meme_chipScore_GM12878$allmotif)
head(meme_chipScore_GM12878$cebpmotif)
p <- ggplot(meme_chipScore_GM12878$cebpmotif, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif: GM12878-MemeChip')
ggsave(file='figs/HudsonAlpha_GM12878_CebpB_Score_CebpMotif.pdf')

p <- ggplot(meme_chipScore_GM12878$allmotif, aes(score, colour=id, group=id)) +
        geom_density()  
ggsave(file='figs/HudsonAlpha_GM12878_CebpB_Score_AllMotif.pdf')

@
\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 3in, height = 4in]{figs/GM12878_meme.png}}
    \\
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/HudsonAlpha_GM12878_CebpB_Score_CebpMotif.pdf}}
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/HudsonAlpha_GM12878_CebpB_Score_AllMotif.pdf}}
    \caption{Distribution of ChIPseq Scores over Motif1: GM12878-Meme-Chip}
    \label{fig:chipseqScoreMotif1}
\end{figure}
\clearpage

\subsection{Meme-Chip:HCT116}
<<MemeHCT116>>=
meme_chipScore_HCT116 <- getCebpAllMotif(peakScore_f=paste0(work_dir,
                                          'Input/HudsonAlpha_HCT116_CebpB_Peaks_0.5kb.bed'), 
                                          allmotif_f=paste0(work_dir,
                                          'Output/ChipMeme_HudsonAlpha_HCT116/Motif/allmotif.msite'),
                                          cebp_motif= 'motif_1')
p <- ggplot(meme_chipScore_HCT116$cebpmotif, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif: HCT116-MemeChip')
ggsave(file='figs/HudsonAlpha_HCT116_CebpB_Score_CebpMotif.pdf')

p <- ggplot(meme_chipScore_HCT116$allmotif, aes(score, colour=id, group=id)) +
        geom_density()  
ggsave(file='figs/HudsonAlpha_HCT116_CebpB_Score_AllMotif.pdf')

@
\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 3in, height = 4in]{figs/HCT116_meme.png}}
    \\
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/HudsonAlpha_HCT116_CebpB_Score_CebpMotif.pdf}}
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/HudsonAlpha_HCT116_CebpB_Score_AllMotif.pdf}}
    \caption{Distribution of ChIPseq Scores over Motif1: GM12878-Meme-Chip}
    \label{fig:chipseqScoreMotif1}
\end{figure}
\clearpage

\subsection{Meme-Chip:K562}
<<MemeK562>>=
meme_chipScore_K562 <- getCebpAllMotif(peakScore_f=paste0(work_dir,
                                          'Input/HudsonAlpha_K562_CebpB_Peaks_0.5kb.bed'), 
                                          allmotif_f=paste0(work_dir,
                                          'Output/ChipMeme_HudsonAlpha_K562/Motif/allmotif.msite'),
                                          cebp_motif= 'motif_1')
p <- ggplot(meme_chipScore_K562$cebpmotif, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif: K562-MemeChip')
ggsave(file='figs/HudsonAlpha_K562_CebpB_Score_CebpMotif.pdf')

p <- ggplot(meme_chipScore_K562$allmotif, aes(score, colour=id, group=id)) +
        geom_density()  
ggsave(file='figs/HudsonAlpha_K562_CebpB_Score_AllMotif.pdf')

@
\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 3in, height = 4in]{figs/K562_meme.png}}
    \\
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/HudsonAlpha_K562_CebpB_Score_CebpMotif.pdf}}
    \subfloat{\includegraphics[width = 3in, height = 3in]{figs/HudsonAlpha_K562_CebpB_Score_AllMotif.pdf}}
    \caption{Distribution of ChIPseq Scores over Motif1: K562-Meme-Chip}
    \label{fig:chipseqScoreMotif1}
\end{figure}
\clearpage

\section{RSAT}
<<DataPreProcessingMotif, cache=TRUE>>=


rsat_GM12878 <- getCebpNon_rsat(input_f=paste0(work_dir, 
    'Output/Rsat/HudsonAlpha_GM12878_CebpB_Peaks_1kb/results/sites/peak_motifs_sites.tab'),
                               peak_f=paste0(work_dir, 
                                          'Input/HudsonAlpha_GM12878_CebpB_Peaks_1kb.bed'), 
                               jaspar_f=paste0(work_dir, 
    'Output/Rsat/HudsonAlpha_GM12878_CebpB_Peaks_1kb/results',
    '/discovered_vs_db/peak-motifs_db_jaspar.tab'))

rsat_HCT116 <- getCebpNon_rsat(input_f=paste0(work_dir, 
    'Output/Rsat/HudsonAlpha_HCT116_CebpB_Peaks_1kb/results/sites/peak_motifs_sites.tab'),
                               peak_f=paste0(work_dir, 
                                          'Input/HudsonAlpha_HCT116_CebpB_Peaks_1kb.bed'), 
                               jaspar_f=paste0(work_dir, 
    'Output/Rsat/HudsonAlpha_HCT116_CebpB_Peaks_1kb/results',
    '/discovered_vs_db/peak-motifs_db_jaspar.tab'))

rsat_K562 <- getCebpNon_rsat(input_f=paste0(work_dir, 
    'Output/Rsat/HudsonAlpha_K562_CebpB_Peaks_1kb/results/sites/peak_motifs_sites.tab'),
                               peak_f=paste0(work_dir, 
                                          'Input/HudsonAlpha_K562_CebpB_Peaks_1kb.bed'), 
                               jaspar_f=paste0(work_dir, 
    'Output/Rsat/HudsonAlpha_K562_CebpB_Peaks_1kb/results',
    '/discovered_vs_db/peak-motifs_db_jaspar.tab'))

@

<<plotRsat>>=

## GM12878
p <- ggplot(rsat_GM12878$nonCEBP, 
            aes(score, colour=motifName, group=motifName )) +
        geom_density()
ggsave(file='figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat.pdf')

# plot motifs in chunks, each for 10 motifs
plotNMotifScore(rsat_GM12878$nonCEBP, 
                output_f ='figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk', 
                N=5)

p <- ggplot(rsat_GM12878$CEBP, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif:GM12878')
ggsave(file='figs/HudsonAlpha_GM12878_CebpB_Score_CEBP_rsat')


## HCT116
p <- ggplot(rsat_HCT116$nonCEBP, 
            aes(score, colour=motifName, group=motifName )) +
        geom_density()
ggsave(file='figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat.pdf')
# plot motifs in chunks, each for 10 motifs
plotNMotifScore(rsat_HCT116$nonCEBP, 
                output_f ='figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk', 
                N=5)
p <- ggplot(rsat_HCT116$CEBP, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif:HCT116')
ggsave(file='figs/HudsonAlpha_HCT116_CebpB_Score_CEBP_rsat.pdf')

## K562
p <- ggplot(rsat_K562$nonCEBP, 
            aes(score, colour=motifName, group=motifName )) +
        geom_density()
ggsave(file='figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat.pdf')
# plot motifs in chunks, each for 10 motifs
plotNMotifScore(rsat_K562$nonCEBP, 
                output_f ='figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk', 
                N=5)
p <- ggplot(rsat_K562$CEBP, aes(x=score))+ 
        geom_histogram(aes(y = ..density..), binwidth=1, fill='#969696') + 
        geom_line(stat='density', alpha=0.6, size=1.5) + 
        ggtitle('ChIPseq Score Distribution of CEBP Motif:K562')
ggsave(file='figs/HudsonAlpha_K562_CebpB_Score_CEBP_rsat.pdf')



@
\subsection{Rsat:GM12878}
\begin{figure}[hbtp]
    \includegraphics{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat.pdf}
    \caption{Distribution of ChIPseq Scores over all nonCEBP Motifs:GM12878}
    \label{fig:chipseqScoreNonCEBPGM12878}
\end{figure}
\clearpage


\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk1.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk2.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk3.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk4.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk5.pdf}}
    \\
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk6.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk7.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk8.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk9.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk10.pdf}}
    \\
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk11.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk12.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk13.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk14.pdf}}
    \subfloat{\includegraphics[width = 1in, height 1in]{figs/HudsonAlpha_GM12878_CebpB_Score_nonCEBP_rsat_chunk15.pdf}}
    \caption{Distribution of ChIPseq Scores over all Non-CEBP Motifs in Chunks:GM12878}
    \label{fig:chipseqScoreNonCEBPChunksGM12878}
\end{figure}


\subsection{Rsat:HCT116}
\begin{figure}[hbtp]
    \includegraphics{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat.pdf}
    \caption{Distribution of ChIPseq Scores over all nonCEBP Motifs:HCT116}
    \label{fig:chipseqScoreNonCEBPHCT116}
\end{figure}
\clearpage

\begin{figure}[hbtp]
    \includegraphics{figs/HudsonAlpha_HCT116_CebpB_Score_CEBP_rsat.pdf}
    \caption{Distribution of ChIPseq Scores over all CEBP Motifs:HCT116}
    \label{fig:chipseqScoreCEBPHCT116}
\end{figure}
\clearpage


\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk1.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk2.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk3.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk4.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk5.pdf}}
    \\
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk6.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk7.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk8.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk9.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk10.pdf}}
    \\
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk11.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_HCT116_CebpB_Score_nonCEBP_rsat_chunk12.pdf}}
    \caption{Distribution of ChIPseq Scores over all Non-CEBP Motifs in Chunks:HCT116}
    \label{fig:chipseqScoreNonCEBPChunksGM12878}
\end{figure}

\subsection{Rsat:K562}
\begin{figure}[hbtp]
    \includegraphics{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat.pdf}
    \caption{Distribution of ChIPseq Scores over all nonCEBP Motifs:K562}
    \label{fig:chipseqScoreNonCEBPK562}
\end{figure}
\clearpage

\begin{figure}[hbtp]
    \includegraphics{figs/HudsonAlpha_K562_CebpB_Score_CEBP_rsat.pdf}
    \caption{Distribution of ChIPseq Scores over all CEBP Motifs:K562}
    \label{fig:chipseqScoreCEBPK562}
\end{figure}
\clearpage


\begin{figure}[!htp]
    \centering
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk1.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk2.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk3.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk4.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk5.pdf}}
    \\
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk6.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk7.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk8.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk9.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk10.pdf}}
    \\
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk11.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk12.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk13.pdf}}
    \subfloat{\includegraphics[width = 1in, height = 2in]{figs/HudsonAlpha_K562_CebpB_Score_nonCEBP_rsat_chunk14.pdf}}
    \caption{Distribution of ChIPseq Scores over all Non-CEBP Motifs in Chunks:K562}
    \label{fig:chipseqScoreNonCEBPChunksK562}
\end{figure}
\clearpage




\section{Metainfo}
<<>>=
sessionInfo()
@

<<knitIt,cache=TRUE, results='hide', message=FALSE, warning=FALSE>>=
library(knitr)
purl("motifAnalysis.Rnw" ) # compile to tex
purl("motifAnalysis.Rnw", documentation = 0) # extract R code only
knit2pdf("motifAnalysis.Rnw")
@



\end{document}

